vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -47,20 +47,8 @@ def _get_inputs_of_figure_interactions(
     inputs = []
     for action in figure_interactions_on_page:
         triggered_model = model_manager._get_action_trigger(action_id=ModelID(str(action.id)))
-        if isinstance(triggered_model, Table):
-            inputs.append(
-                {
-                    "active_cell": State(
-                        component_id=triggered_model._callable_object_id, component_property="active_cell"
-                    ),
-                    "derived_viewport_data": State(
-                        component_id=triggered_model._callable_object_id, component_property="derived_viewport_data"
-                    ),
-                }
-            )
-        else:
-            inputs.append({"clickData": State(component_id=triggered_model.id, component_property="clickData")})
-
+        if hasattr(triggered_model, "_figure_interaction_input"):
+            inputs.append(triggered_model._figure_interaction_input)
```

Comment:
Since this part of the code is executed in the build app phase and only for the models that have the filter_interaction actions attached, I would raise an exception here:
- if `_figure_interaction_input ` model method isn't defined. 

I would also raise an exception inside the `_apply_filter_interaction` function:
- if "modelID" doesn't exist inside the ctd_filter_interaction, or - (There is a change we will remove this limitation soon)
- if `_filter_interaction` model method isn't defined


This move covers the case when users create customTableComponent and attach filter_interaction action on it. I know it's almost impossible to expect someone will do that, but we should probably cover these cases as well. 
(I'm also fine with skipping solving this case if it's too much work right now)

Reply:
Ok sounds good! I removed the if clause as it's not necessary anymore and added the relevant Errors